ts/test/session/unit/utils/RingBuffer_test.ts

```diff
@@ -0,0 +1,106 @@
+// tslint:disable: no-implicit-dependencies max-func-body-length no-unused-expression no-require-imports no-var-requires
+
+import chai from 'chai';
+import { RingBuffer } from '../../../../session/utils/RingBuffer';
+
+const { expect } = chai;
+
+describe('RingBuffer Utils', () => {
+  it('gets created with right capacity', () => {
+    const ring = new RingBuffer<number>(5000);
+    expect(ring.getCapacity()).to.equal(5000);
+    expect(ring.getLength()).to.equal(0);
+    expect(ring.has(0)).to.equal(false, '4 should not be there');
+  });
+
+  describe('length & capacity are right', () => {
+    it('length is right 1', () => {
+      const ring = new RingBuffer<number>(4);
+      ring.add(0);
+      expect(ring.getLength()).to.equal(1);
+    });
+
+    it('length is right 4', () => {
+      const ring = new RingBuffer<number>(4);
+      ring.add(0);
+      ring.add(1);
+      ring.add(2);
+      ring.add(3);
+      expect(ring.getLength()).to.equal(4);
+    });
+
+    it('capacity does not get exceeded', () => {
+      const ring = new RingBuffer<number>(4);
+      ring.add(0);
+      ring.add(1);
+      ring.add(2);
+      ring.add(3);
+      ring.add(4);
+      expect(ring.getLength()).to.equal(4);
+    });
+  });
+
+  it('items are removed in order 1', () => {
+    const ring = new RingBuffer<number>(4);
+    ring.add(0);
+    ring.add(1);
+    ring.add(2);
+    ring.add(3);
+    ring.add(4);
+    expect(ring.has(0)).to.equal(false, '0 should not be there anymore');
+    expect(ring.has(1)).to.equal(true, '1 should still be there');
+    expect(ring.has(2)).to.equal(true, '2 should still be there');
+    expect(ring.has(3)).to.equal(true, '3 should still be there');
+    expect(ring.has(4)).to.equal(true, '4 should still be there');
+  });
+
+  it('two times the same items can exist', () => {
```

Comment:
Does this mean that the buffer can have duplicates stored in different places? There is no merging?

Reply:
yeah, a ring buffer does not care about what is stored. 
I kept it like this because 
1. this is what a basic ring buffer should do, and if we want to reuse this somewhere else, we need to work the default way.
2. we should not get two times the same deleted message from the API
3. even if we get the same id two times, that doesn't break the logic.

Reply:
I see. 👍